pathiomock.go

```diff
@@ -0,0 +1,59 @@
+package pathio
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sync"
+)
+
+// MockClient mocks out an S3 bucket
```

Comment:
Agreed, just wanted to come to a consensus on the mock interface prior to writing those docs.